scripts/tests/run_tv_casting_test.py

```diff
@@ -82,20 +107,79 @@ class LogValueExtractor:
     This class provides a centralized way to extract values from log lines and manage the error handling and logging process.
     """
 
-    def __init__(self, casting_state: str, log_paths: List[str]):
-        self.casting_state = casting_state
+    def __init__(self, log_paths: List[str]):
         self.log_paths = log_paths
 
     def extract_from(self, line: str, value_name: str):
         if value_name in line:
             try:
-                return extract_value_from_string(line, value_name, self.casting_state, self.log_paths)
+                return extract_value_from_string(line, value_name, self.log_paths)
             except ValueError:
                 logging.error(f'Failed to extract `{value_name}` value from line: {line}')
-                handle_casting_failure(self.casting_state, self.log_paths)
+                handle_casting_failure(self.log_paths)
         return None
 
 
+class CommissionerNumberHandler:
+    """A class to manage updating the commissioner number placeholder.
+
+    This class provides functionality to update the commissioner number placeholder in an input command with the actual valid discovered commissioner number.
+    """
+
+    INVALID_COMMISSIONER_NUMBER = '-1'
+
+    def __init__(self):
+        self._current_commissioner_number = self.INVALID_COMMISSIONER_NUMBER
+
+    @property
+    def commissioner_number(self) -> str:
+        return self._current_commissioner_number
+
+    @commissioner_number.setter
+    def commissioner_number(self, new_value: str):
+        self._current_commissioner_number = new_value
+
+    def is_valid(self) -> bool:
+        return self._current_commissioner_number != self.INVALID_COMMISSIONER_NUMBER
+
+
+class Step:
+    """A class to represent a step in a test sequence for validation.
+
+    A Step object contains attributes relevant to a test step.
```

Comment:
Can we represent _**starting**_ and _**stopping**_ the apps using `Steps`?

Reply:
I don't think it would be possible to represent the _**starting**_ the apps using `Steps` since we start the subprocesses with the _ProcessManager_. The _ProcessManager_ also takes care of stopping the subprocesses once we are done with them.

Reply:
Not sure I followed why this isn't possible? Can we have reserved input\_cmd keywords for "Start" and "Stop" that can run the code to call the ProcessManager?

Reply:
Not sure I followed why this is not possible. Can't we define reserved input\_cmd keywords for "Start" and "Stop" that run the ProcessManager code we need to run?

Reply:
Addressed in the latest commit.